addons/fire/functions/fnc_burn.sqf

```diff
@@ -144,6 +144,10 @@ if (_isBurning) exitWith {};
     };
     
     if !(isGamePaused) then {
+        if (_isThisUnitAlive) then {
+            _isThisUnitAlive = (!(isNull _unit) && { !(_unit getVariable [QGVAR(killed), false]) });
```

Comment:
I think I don't get this. Why you're doing this instead of just using `alive` ?

Also when using `alive` there's no reason to check `!isNull _obj && alive _obj` as `alive objNull` is false.

Reply:
If the unit goes to spectator they will report `alive _unit` as true and they will still be on fire and will still take damage. This is the only work around I could think of, and even then I'm not too sure if it works

Reply:
Add a comment then.